source/Calamari/Kubernetes/Conventions/HelmUpgradeConvention.cs

```diff
@@ -69,11 +69,10 @@ string BuildHelmCommand(RunningDeployment deployment, ScriptSyntax syntax)
 
             var customHelmExecutable = CustomHelmExecutableFullPath(deployment.Variables, deployment.CurrentDirectory);
 
-            AssertHelmV3(customHelmExecutable);
-
             var sb = new StringBuilder();
 
             SetExecutable(sb, syntax, customHelmExecutable);
+            AssertHelmV3(customHelmExecutable);
```

Comment:
`SetExecutable` makes the binary executable (if it's a custom one)
`AssertHelmV3` calls the helm binary

So this would always fail if the binary wasn't executable